openshift_resources/oadp/data_protection_application.py

```diff
@@ -1,17 +1,516 @@
+        None,
+        description='nodeSelector defines the nodeSelector to be supplied to Restic podSpec',
+    )
+    resourceAllocations: Optional[Any] = Field(
+        None,
+        description='resourceAllocations defines the CPU and Memory resource allocations for the restic Pod',
+    )
+    tolerations: Optional[List[TolerationModel]] = Field(
+        None,
+        description='tolerations defines the list of tolerations to be applied to Restic daemonset',
+    )
+
+
+class DataProtectionApplicationConfigurationVelero(BaseModel):
+    customPlugins: Optional[List[CustomPlugin]] = Field(
+        None,
+        description='customPlugins defines the custom plugin to be installed with Velero',
+    )
+    defaultPlugins: Optional[List[str]] = None
+    featureFlags: Optional[List[str]] = Field(
+        None,
+        description='featureFlags defines the list of features to enable for Velero instance',
+    )
+    logLevel: Optional[LogLevel] = Field(
+        None,
+        description='Velero server’s log level (use debug for the most logging, leave unset for velero default)',
+    )
+    noDefaultBackupLocation: Optional[bool] = Field(
+        None,
+        description='If you need to install Velero without a default backup storage location noDefaultBackupLocation flag is required for confirmation',
+    )
+    podConfig: Optional[DataProtectionApplicationConfigurationVeleroPodConfig] = Field(
+        None, description='Pod specific configuration'
+    )
+    restoreResourcesVersionPriority: Optional[str] = Field(
+        None,
+        description='restoreResourceVersionPriority represents a configmap that will be created if defined for use in conjunction with EnableAPIGroupVersions feature flag Defining this field automatically add EnableAPIGroupVersions to the velero server feature flag',
+    )
+
+
+class DataProtectionApplicationConfiguration(BaseModel):
+    restic: Optional[Restic] = Field(
+        None, description='ResticConfig is the configuration for restic server'
+    )
+    velero: Optional[DataProtectionApplicationConfigurationVelero] = None
+
+
+class DataMover(BaseModel):
+    credentialName: Optional[str] = Field(
+        None, description='User supplied Restic Secret name'
+    )
+    enable: Optional[bool] = Field(
+        None,
+        description='enable flag is used to specify whether you want to deploy the volume snapshot mover controller',
+    )
+    timeout: Optional[str] = Field(
+        None,
+        description='User supplied timeout to be used for VolumeSnapshotBackup and VolumeSnapshotRestore to complete, default value is 10m',
+    )
+
+
+class Features(BaseModel):
+    dataMover: Optional[DataMover] = Field(
+        None, description='Contains data mover specific configurations'
+    )
+
+
+class Option(BaseModel):
+    name: Optional[str] = Field(None, description='Required.')
+    value: Optional[str] = None
+
+
+class PodDnsConfig(BaseModel):
```

Comment:
PodDnsConfig is defined in core.v1: https://github.com/oadp-qe/openshift-resources/blob/main/src/resources/io/k8s/api/core/v1.py#L896
, please remove and import it from there instead